MANIFEST.in

```
@@ -0,0 +1 @@
+graft extlib
```

Don't we need C++ source files? `libasd` directory has headers and `python` directory has source code for python binding.

```suggestion
graft extlib
graft python
graft libasd
include LICENSE CMakeLists.txt
```

---

AFAICS all these folders already get included by default (this can be checked by unpacking the sdist).  OTOH it probably doesn't hurt to explicitly list them.

---

Actually, I already ran `python3 setup.py sdist` and untar-ed the `dist/libasd-1.5.5.tar.gz` file before I commented it. And I found that, with the simple `MANIFEST.in`, neither `python` nor `libasd` were included. As you know, this library is fully written in C++, it wouldn't compile without C++ sources. Since those directories are specific to my repository, there is no reason that those directories, especially for `libasd` directory which only includes C++-related files, are included by default. If all of the directories will be included by default, we don't need `MANIFEST.in`. FYI, my environment is Ubuntu 20.04 and python 3.8.

Can you apply the changes I suggested here?

---

I guess that may depend on the version of setuptools :-(  In any case I added the lines you suggested, as they certainly cannot hurt (and as you suggest may be necessary sometimes).

To be clear, I think the correct way to check whether the sdist is complete is to try to install it or to build a wheel from it; this will fail if files are missing.

---

Actually, I also ran `python3 setup.py build` in the untar-ed `dist/libasd-1.5.5` that is created with the `MANIFEST.in` and it fails because it does not have `CMakeLists.txt` nor C++ sources. I didn't mention that because it is obvious that it fails to build if there is no source code.

The version of `setuptools` installed via pip is `45.2.0`. That is really older than the latest version, but is the default version on Ubuntu 20.04. Considering that we will run `setuptools` mostly on CI, I think it is good to support default version on a distro that is common in CI service.